contracts/service-registry/src/contract/execute.rs

```diff
@@ -256,3 +244,15 @@ pub fn claim_stake(
         .to_vec(),
     }))
 }
+
+fn ensure_service_exists(
+    storage: &dyn Storage,
+    service_name: &String,
+) -> Result<(), ContractError> {
+    ensure!(
+        state::has_service(storage, service_name),
+        ContractError::ServiceNotFound
+    );
```

Comment:
I like this change to call has_service, but I don't think extracting the outer function has a lot of benefits, it adds more lines than it saves 😄

Reply:
I decided to do it this way, because I noticed some previous inconsistencies between all functions when checking for service and reporting errors. This way enforces a standard error